### PR TITLE
Split the finalizer and the infra-id

### PR DESCRIPTION
**Pull request by jnpacker**
Split the infra-id and the finalizer
* This was needed to account for Cluster Templates that we are proto-typing.
* Fixed the deploy command
* Updated the documentation slightly

Signed-off-by: Joshua Packer <jpacker@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,7 +100,7 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 
 .PHONY: deploy
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	#cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
```
Comment (zhujian7) [inline Makefile:103]:
```suggestion
	cd config/deployment && $(KUSTOMIZE) edit set image quay.io/stolostron/hypershift-deployment-controller:latest=${IMG}
```

Comment (zhujian7) [reply Makefile:103]: And give the `REPO` a default value `quay.io/stolostron/` in Makefile.

Comment (jnpacker) [reply Makefile:103]: I'm addressing this in the new Basic Tests pull request. closing this pull request.
@zhujian7 I added the REPO URL you suggested and uncommented the line.

**Comment by jnpacker** [comment]
This code is combined into this pull request: https://github.com/stolostron/hypershift-deployment-controller/pull/16

**Pull request closed without merging**